aocd/get.py

```diff
@@ -157,17 +165,17 @@ def giveup(msg):
         log.debug("non-interactive")
         raise giveup("Failed introspection of filename")
     years = {int(year) for year in re.findall(pattern_year, abspath)}
-    if len(years) > 1:
+    if len(years) != 1:
         raise giveup("Failed introspection of year")
-    year = years.pop() if years else None
+    year = years.pop()
```

Comment:
Unfortunately this stricter parsing is no bueno - I have it written in the README [here](https://github.com/wimglenn/advent-of-code-data#how-does-this-library-work) that a filename like `q03.py` would work, and it works by returning `year=None` then letting some later code check for `None` and call `most_recent_year()` instead. 

Even though the behavior is questionable (the aocd introspection will only work currently when run within the same year as the event, and those scripts will start to break a week after the event finishes), I think it's best to keep this behavior of being able to omit mentioning the current year in your filenames entirely.